django/contrib/postgres/forms/array.py

```diff
@@ -202,17 +214,19 @@ def clean(self, value):
                 cleaned_data.append(None)
             else:
                 errors.append(None)
-        if self.remove_trailing_nulls:
-            null_index = None
-            for i, value in reversed(list(enumerate(cleaned_data))):
-                if value in self.base_field.empty_values:
-                    null_index = i
-                else:
-                    break
-            if null_index is not None:
-                cleaned_data = cleaned_data[:null_index]
-                errors = errors[:null_index]
+        cleaned_data, null_index = self._remove_trailing_nulls(cleaned_data)
+        if null_index is not None:
+            errors = errors[:null_index]
         errors = list(filter(None, errors))
         if errors:
             raise ValidationError(list(chain.from_iterable(errors)))
         return cleaned_data
+
+    def has_changed(self, initial, data):
+        try:
+            data, _ = self._remove_trailing_nulls(self.to_python(data))
+        except ValidationError:
+            pass
```

Comment:
Since `self._remove_trailing_nulls` is not meant to `raise ValidationError` I'd only call `to_python` in the `try` block

```suggestion
        try:
            data = self.to_python(data)
        except ValidationError:
            pass
        else:
            data = self._remove_trailing_nulls(data)[0]
```

Reply:
Makes sense. Fixed